Aufgabe 2/Codes/JS/uno_code2.ts

```diff
@@ -1,44 +1,58 @@
 namespace unogame2 {
-    interface Card {
-        color: string;
-        value: string;
+  interface Card {
+    color: string;
+    value: string;
+  }
+
+  let colors: string[] = ["Red", "Blue", "Yellow", "Green"];
+  let values: string[] = [
+    "0",
+    "1",
+    "2",
+    "3",
+    "4",
+    "5",
+    "6",
+    "7",
+    "8",
+    "9",
+    "+2",
+    "X",
+    "RW"
+  ];
+  const handCards: Card[] = [];
+  const deck: Card[] = [];
+
+  let card: Card;
+  for (let i: number = 0; i < colors.length; i++) {
+    for (let a: number = 0; a < values.length; a++) {
+      for (let z: number = 0; z < 2; z++) {
+        card = { color: colors[i], value: values[a] };

```

Comment:
Achte darauf, welche Zählvariable du wo als Arrayindex verwendest. Außerdem musst du auch beachten wann du deine Karte erstellst bzw. in das Array abspeicherst.